enhancements/baremetal/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,188 @@
+---
+title: Baremetal IPI Config Enhancement
+authors:
+  - "@sadasu"
+reviewers:
+  - "@deads2k"
+  - "@hardys"
+  - "@enxebre"
+  - "@squeed"
+  - "@abhinavdahiya"
+  - "@dhellmann"
+
+approvers:
+  - "@abhinavdahiya"
+  - "@enxebre"
+  - "@deads2k"
+
+creation-date: 2019-11-14
+last-updated: 2019-11-26
+status: not implemented
+see-also:
+  - 
+replaces:
+  - https://github.com/openshift/enhancements/pull/90
+superseded-by:
+  - 
+---
+
+# Config required for Baremetal IPI deployments
+
+## Release Signoff Checklist
+
+- [*] Enhancement is `implementable`
+- [*] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions
+
+1. Which is a preferred name for the new CR "Metal3ProvisioningController" or
```

Comment:
Hey @sadasu thanks for putting this together.  I'm confused why controller?, unless I'm getting it wrong this is purely a config right?
so this should be the likes of `baremetal.config/operator.openshift.io` or `metal3.config/operator.openshift.io`
The main questions to be fleshed out are:
who installs the crd?
who instantiate the object for that crd?
who reads that object?
Does its values ever change?
Can we please keep the discussion in one single place and try to come up with a final solution here https://github.com/openshift/enhancements/pull/90#issuecomment-559804188

Reply:
I asked @abhinavdahiya yesterday to check the latest discussion on #90 so hopefully he will do so soon, but FWIW he also made it clear that he preferred this approach, so we've been trying to reach consensus to unblock the implementation.  

Over various PRs this discussion has been ongoing since August, so IMHO we just need to pick and interface and get this done.

To answer your questions:

- Yes this is purely config, maybe the CR name should be different - Metal3ConfigData?  I'm not sure what the conventions are here so feel free to suggest an appropriate name.

- Other metal3 CRDs are installed by the MAO, so I guess this will be the same @sadasu ?

- The CR object will be created by the installer, based on data parsed from the install-config

- The object will be read by the MAO, and perhaps in future the MCO, to influence the configuration of the provisioning containers in the metal3 pod (started by the MAO), and perhaps the keepalived configuration created by the MCO cc @celebdor 

- Some values could change, for example to increse the size of the configured DHCP pool

Happy to continue discussion on #90 but it seemed appropriate to reply to those specific questions here :)

Reply:
as for naming.

let's make sure the name provides the intuition that it is used to configure the provisioning network + it's services.

for who will create the CRD, I think all the CRDs for MAO are created using CVO so this wouldn't be any different.

for who will create the CR, i think currently the plan is for the installer to push this object.

> I'm confused why controller?, unless I'm getting it wrong this is purely a config right?

I think almost all CRs are configs for controllers.. so this isn't any different.